testworkload/nginx_workload.go

```diff
@@ -58,46 +58,36 @@ func NewNginx(kc kubernetes.Interface, namespace string, options ...NginxOpts) (
 		},
 		client: kc,
 	}
-
 	//apply options
 	for _, option := range options {
 		if err := option(n); err != nil {
 			return nil, fmt.Errorf("error invalid options: %v", err)
 		}
 	}
 
-	//create nginx deployment
```

Comment:
I would move create() calls out of the poll loop (and create the service and deployment together). Client-go already has built-in retries for create calls, no need to duplicate that. I'd actually fail hard if the deployment already exists.

Then I'd have your two checks in a poll loop. Not sure why you need the checks (you're validating that kubernetes does its job?). But those are loop-worthy.

One way to organize this would be to have a `var err error` at the top, and `defer` an anonymous cleanup function to delete if `err` is non-nil.

Then you just bail at any error and cleanup will always get called. The only trick is to be careful with the scope of err, since it's easy to override it.